doc/examples/edges/plot_skeleton.py

```diff
@@ -47,18 +43,146 @@
 skeleton = skeletonize(image)
 
 # display results
-fig, (ax1, ax2) = plt.subplots(nrows=1, ncols=2, figsize=(8, 4.5),
-                               sharex=True, sharey=True,
-                               subplot_kw={'adjustable': 'box-forced'})
+fig, axes = plt.subplots(nrows=1, ncols=2, figsize=(8, 4),
+                         sharex=True, sharey=True,
+                         subplot_kw={'adjustable': 'box-forced'})
 
-ax1.imshow(image, cmap=plt.cm.gray)
-ax1.axis('off')
-ax1.set_title('original', fontsize=20)
+ax = axes.ravel()
 
-ax2.imshow(skeleton, cmap=plt.cm.gray)
-ax2.axis('off')
-ax2.set_title('skeleton', fontsize=20)
+ax[0].imshow(image, cmap=plt.cm.gray)
+ax[0].axis('off')
+ax[0].set_title('original', fontsize=20)
+
+ax[1].imshow(skeleton, cmap=plt.cm.gray)
+ax[1].axis('off')
+ax[1].set_title('skeleton', fontsize=20)
+
+fig.tight_layout()
+plt.show()
+
+######################################################################
+# **skeletonize vs skeletonize 3d**
+#
+# ``skeletonize`` [Zha84]_ works by making successive passes of
+# the image, removing pixels on object borders. This continues until no
+# more pixels can be removed.  The image is correlated with a
+# mask that assigns each pixel a number in the range [0...255]
+# corresponding to each possible pattern of its 8 neighbouring
+# pixels. A look up table is then used to assign the pixels a
+# value of 0, 1, 2 or 3, which are selectively removed during
+# the iterations.
+#
+#  ``skeletonize_3d`` [Lee94]_ uses an octree data
+# structure to examine a 3x3x3 neighborhood of a pixel. The algorithm
+# proceeds by iteratively sweeping
+# over the image, and removing pixels at each iteration until the image
+# stops changing. Each iteration consists of two steps: first, a list of
+# candidates for removal is assembled; then pixels from this list are
+# rechecked sequentially, to better preserve connectivity of the image.
+
+import numpy as np
+from scipy import ndimage as ndi
+from skimage.morphology import skeletonize, skeletonize_3d
+import matplotlib.pyplot as plt
+
+
+def microstructure(lsize=256):
```

Comment:
If you refactor the example, I think this function could be removed since we now have ``data.binary_blobs``